packages/driver/src/baseConn.ts

```diff
@@ -134,6 +140,9 @@ export class BaseRawConnection {
 
   protected stateCodec: ICodec = INVALID_CODEC;
   protected stateCache: [Session, Uint8Array] | null = null;
+  protected lastStateUpdate: any = null;
```

Comment:
Unless we need to use it in runtime code, my vote is for `unknown` since it's the simplest possible safe type here. Otherwise, if there _is_ a common interface for this property that we _might_ want to use in the future, now seems like a good time to give this a name while we're in here.

And, to reiterate, my preference for those record types would be `Record<string, unknown>` rather than `{ [name: string]: any }` just to require downstream checking if anything ever did want to access a property.

Reply:
Yep, good point. It does share the type with the session state we send, which is currently typed as `any`. And aside from changing `any` -> `unknown` which makes sense, is there a reason to use `Record<Key, Type>` over `{[key: Key]: Type}`? For me the second seems more useful, since it can have a hint of what the key represents, eg. is it a mapping of 'names' or 'ids' to something.

Reply:
Oh, sure, I just thought `name` was a throwaway identifier 😆. I mostly prefer the Record syntax because it communicates to me that it's just a record rather than having some hidden mapped type magic that I should check for, so makes it a little easier to glance at. Not a big deal for me either way.